onnxruntime/core/providers/js/operators/conv.h

```diff
@@ -43,36 +43,36 @@ class ConvBase : public JsKernel {
                                    "dilations" : [$2],
                                    "group" : $3,
                                    "kernel_shape" : [$4],
-                                   "pads" : $5 ? Array.from(HEAP32.subarray($6, $6 + $5)) : [],
+                                   "pads" : $5 ? Array.from(HEAP32.subarray($6 >>> 0, ($6 >>> 0) + $5)) : [],
```

Comment:
After done some local testing, we know that Emscripten will append ">>>0" into parameters like $5, $6, so no need to put it here. (Otherwise the generated code will be like `   "pads": $5 ? Array.from(HEAP32.subarray($6 >>> 0 >>> 0, ($6 >>> 0) + $5 >>> 0)) : [],` )

As a result, I think the best practice is, **do not calculate offset inside any embedded JS**. We may need to change code like:
```js
"pads" : $5 ? Array.from(HEAP32.subarray($6, $6 + $5)) : [],
...
, length, data_ptr,
```

into

```js
"pads" : $5 ? Array.from(HEAP32.subarray($5, $6)) : [],
...
, data_ptr_start, data_ptr_end,
```

Reply:
When we make sure we don't directly calculate parameters, Emscripten will just append that ">>>0" and work fine.

Reply:
fair, we can move the ptr + size to the c++ side

Reply:
not so easy to use start_ptr and end_ptr because we also need the size on the javascript side to determine if we need an empty array, ie 
"outputPadding" : **$10** ? Array.from(HEAP32.subarray($11 >>> 0, ($11 >>> 0) + $10)) : [],

Reply:
ok, I think this is now pretty clean - take a look